Dockerfiles/agent-ot/Dockerfile

```diff
@@ -0,0 +1,28 @@
+ARG BASE_IMAGE_DD_VERSION=latest
+FROM docker.io/datadog/agent:$BASE_IMAGE_DD_VERSION AS baseimage
```

Comment:
In CI you don't want to pull from Docker as the image does not exist just yet, you want to use the local reference. The whole `FROM` can be arg instead of just the tag.

Reply:
Thanks for the review! This is actually not for CI builds but for customers to be able to easily produce their own custom builds of the otel-agent based off their modified `otel-agent` stuff. So one would assume that they're pulling from docker images available `docker.io/datadog/agent:${VERSION}` to put their own custom binary on top of it. 

The workflow makes sense to me. I've removed the extraneous `ARG CIBUILD` which is absolute unnecessary. 

Let me know if that makes sense @vboulineau.

Reply:
I see, in that case you can use the default public registry instead: `gcr.io/datadoghq/agent`.